R/proj.R

```diff
@@ -4,6 +4,7 @@ proj_crit <- function() {
   rprojroot::has_file(".here") |
     rprojroot::is_rstudio_project |
     rprojroot::is_r_package |
+    rprojroot::is_git_root |
```

Comment:
I guess this was an oversight? I hadn't noticed till now but the lack of this caused some really puzzling results in my manual tests, until I figure it out. I've always been assuming that git repos would be seen as a project in the `proj_get()/proj_set()` sense here.

Reply:
I think this was deliberate, but I forgot why 😬

Reply:
Maybe it had to do with some form of deliberate nesting? I added it because we need to be able to recognize a newly cloned repo as a project, even if it's not a package. Alternatively, I could set a `.here` sentinel in that case 🤔

Decision: I leave `rprojroot::is_git_root` in here. The universe will remind us if/why that is a bad idea.